soundbay/data.py

```diff
@@ -540,8 +544,14 @@ def _create_start_times(self, filepath: Path) -> np.ndarray:
             audio_dict contains references to audio paths given name from metadata
         """
         audio_len = sf.info(filepath).duration
+        step = self.seq_length * (1-self.overlap)
         decimal_place = abs(Decimal(str(self.seq_length)).as_tuple().exponent)
```

Comment:
shouldn't the `decimal_place` be calculated with respect to the `step` instead of `self.seq_length` now that the starttimes vector is a function of `step`?
not sure what edge case it should catch to begin with actually but seems like it should be changed too here.

Reply:
you are right, but actually i think i should delete the decimal_place and rounding, because its role is to handle the last element, however, since `step` could be shorter than` self.seq_length` (i.e., the range can end after `audio_len` - `self.seq_length`) i need to handle the last element anyway (with the filtering and the addition of the last start time). sorry for only noticing this now.